libbeat/reader/readjson/json.go

```diff
@@ -119,6 +127,66 @@ func createJSONError(message string) common.MapStr {
 	return common.MapStr{"message": message, "type": "json"}
 }
 
+func (pp *JSONPostProcessor) Name() string {
+	return "json"
+}
+
+func (pp *JSONPostProcessor) PostProcess(msg *reader.Message) {
+	jsonFields, ok := msg.Fields[pp.Name()].(common.MapStr)
+	if !ok {
+		return
+	}
+
+	// The message key might have been modified by multiline
+	if len(pp.cfg.MessageKey) > 0 && len(msg.Content) > 0 {
+		jsonFields[pp.cfg.MessageKey] = string(msg.Content)
+	}
```

Comment:
Which value has `MessageKey` and what do we store in here if we didn't run multiline?

Reply:
`MessageKey` is only needed if `multiline` is configured before the JSON parser.